users/tests/test_user_login.py

```diff
@@ -22,3 +33,20 @@ def test_user_client_should_not_log_in_when_email_is_none(self):
 
     def test_user_client_should_not_log_in_when_password_is_none(self):
         self.assertFalse(self.client.login(email="testuser@codepoets.it", password=None))
+
+    def test_user_should_redirect_to_change_password_form_if_need_change_password_is_true(self):
+        data = {"username": self.user.email, "password": self.password}
+        response = self.client.post(reverse("login"), data=data)
+        self.assertRedirects(response, reverse("password_change_on_login"), status_code=302, target_status_code=200)
+
+    def test_user_should_redirect_to_home_if_need_change_password_is_false(self):
+        data = {"username": self.user.email, "password": self.password}
+        self.user.need_change_password = False
+        self.user.save()
+        response = self.client.post(reverse("login"), data=data)
+        self.assertRedirects(response, reverse("home"), status_code=302, target_status_code=200)
+
+    def test_user_with_correct_email_and_password_should_log_in_on_post(self):
+        data = {"username": self.user.email, "password": self.password}
```

Comment:
The same data is repeated 3 times in changes which you added. Please move this to `setUp` method and reuse `self.data` in every case which you need

Reply:
Done.